lnwallet/commitment.go

```diff
@@ -137,6 +137,36 @@ func DeriveCommitmentKeys(commitPoint *btcec.PublicKey,
 	return keyRing
 }
 
+// ScriptInfo holds a redeem script and hash.
+type ScriptInfo struct {
+	// PkScript is the outputs' PkScript.
+	PkScript []byte
+
+	// WitnessScript is the full script required to properly redeem the
+	// output with PkSript. This field will only be populated if a PkScript
+	// is p2wsh or p2sh.
+	WitnessScript []byte
+}
+
+// CommitScriptToRemote creates the script that will pay to the non-owner of
+// the commitment transaction, adding a delay to the script based on the
+// channel type.
+func CommitScriptToRemote(_ channeldb.ChannelType, csvTimeout uint32,
+	key *btcec.PublicKey) (*ScriptInfo, error) {
```

Comment:
Can this then be a method on `CommitmentBuilder`, or also not?

Reply:
Ok, also called from the naked `CreateCommitTx` function. Still wondering if it wouldn't be better eventually to also instantiate a commitment builder in the funding flow.

Reply:
Leaving for later